src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1898,10 +1898,15 @@ private function registerLockConfiguration(array $config, ContainerBuilder $cont
             // Generate stores
             $storeDefinitions = [];
             foreach ($resourceStores as $resourceStore) {
-                $storeDsn = $container->resolveEnvPlaceholders($resourceStore, null, $usedEnvs);
+                if (\is_string($resourceStore) && !empty($resourceStore) && '@' === $resourceStore[0]) {
+                    $storeDsn = new Reference(substr($resourceStore, 1));
+                } else {
+                    $storeDsn = $container->resolveEnvPlaceholders($resourceStore, null, $usedEnvs);
+                }
+
                 $storeDefinition = new Definition(interface_exists(StoreInterface::class) ? StoreInterface::class : PersistingStoreInterface::class);
                 $storeDefinition->setFactory([StoreFactory::class, 'createStore']);
-                $storeDefinition->setArguments([$resourceStore]);
+                $storeDefinition->setArguments([$storeDsn]);
```

Comment:
I'm gonna assume this was a bug up front.... We were processing env vars but never passing the processed output into store factory. Fixed that as part of this.

Reply:
I see. So StoreFactory supports passing in quite a few types, but there's no way to reference anything, the current config is limited to passing a string. 

So far the best I've come up with is a compiler pass or redefining the service definitions for lock. Can we figure out a way to pass a reference too?